requirements.txt

```diff
@@ -0,0 +1,3 @@
+boto
+botocore>=1.10.0
+boto3>=1.0.0
```

Comment:
@jillr :

I feel like we may want to set the bar higher than this for boto3/botocore.
We're just before the Ansible 2.10 cut-off which feels like a good time to do this.

Reply:
For background, here's my data from module data scraping:

https://github.com/AlanCoding/collection-dependencies-demo/blob/master/sniff_req/discovered.json#L43-L49

All entries there are from some module or another. In this case, it appears to come from here:

https://github.com/ansible-collections/amazon.aws/blob/a2b775ece0b3f46d44bfd204a01d36d9fe6a0232/plugins/modules/cloudformation_info.py#L18-L20

That requirement is probably ancient and not really accurate, I just didn't have any other source for a hard version minimum.

Reply:
botocore and boto3 no longer support py2.6, the AWS collections should be python >= 2.7.

For boto3 rev we'll need to find the newest minimum functional version, there's definitely going to be modules that won't work with 1.0.0.  Offhand I don't have a better way of doing this than just testing backwards through a number of versions until something breaks.

Reply:
At some point ago, the minimum version of botocore necessary for any module using the common connection code in `boto3_conn` was `1.4.1`. That corresponds to `boto3>=1.2.6` for the baseline.

Reply:
Lowest we currently test against
boto3_version": "1.9.249", "botocore_version": "1.12.249" 
boto-2.49.0

Reply:
@tremble that is the latest version of `boto`, but it was released July 2018, so that's still giving plenty of leeway IMO.

For building execution environments, I don't think we will be able to support python2. Requiring newer versions also isn't expected to give any trouble (but we will always have to just run and see).